frame/uniques/src/types.rs

```diff
@@ -91,32 +91,32 @@ pub struct InstanceDetails<AccountId, DepositBalance> {
 	pub(super) deposit: DepositBalance,
 }
 
-#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo)]
+#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo, MaxEncodedLen)]
 #[scale_info(skip_type_params(StringLimit))]
-pub struct ClassMetadata<DepositBalance, StringLimit: Get<u32>> {
+pub struct ClassMetadata<DepositBalance, BoundedString> {
 	/// The balance deposited for this metadata.
 	///
 	/// This pays for the data stored in this struct.
 	pub(super) deposit: DepositBalance,
 	/// General information concerning this asset. Limited in length by `StringLimit`. This will
 	/// generally be either a JSON dump or the hash of some JSON which can be found on a
 	/// hash-addressable global publication system such as IPFS.
-	pub(super) data: BoundedVec<u8, StringLimit>,
+	pub(super) data: BoundedString,
```

Comment:
Eh? I'm suspecting this was changed to avoid annotating the `mel_bound` attribute?

Reply:
Yeah, there was more than just the obvious errors appearing. generally i find them very disorienting, especially when this works perfectly fine.

try it out and let me know if you see anything i dont.

Reply:
The `mel_bound` requires you to list out all the type parameters that are supposed to be bound by `MaxEncodedLen`, and so it simply needs `DepositBalance: MaxEncodedLen` in this case, and all will be good.

Reply:
9c50192a384fb6af73e211f469c6e982c68d90d3